kernel/kernel-api/src/test/scala/io/delta/kernel/SnapshotManagerSuite.scala

```diff
@@ -93,4 +103,689 @@ class SnapshotManagerSuite extends AnyFunSuite {
+      val corruptedCheckpoint = FileNames.checkpointFileWithParts(logPath, corruptedVersion, 5)
+        .asScala
+        .map(p => FileStatus.of(p.toString, 10, 10))
+        .take(4)
+      val checkpoints = singularCheckpointFileStatuses(validVersions)
+      val deltas = deltaFileStatuses(deltaVersions)
+      val logSegmentOpt = new SnapshotManager().getLogSegmentForVersion(
+        logPath,
+        createMockTableClient(listFromFileList(deltas ++ corruptedCheckpoint ++ checkpoints)),
+        Optional.empty(),
+        Optional.empty()
+      )
+      val checkpointVersion = validVersions.sorted.lastOption
+      assert(logSegmentOpt.isPresent())
+      checkLogSegment(
+        logSegment = logSegmentOpt.get(),
+        expectedVersion = deltaVersions.max,
+        expectedDeltas = deltaFileStatuses(
+          deltaVersions.filter(_ > checkpointVersion.getOrElse(-1L))),
+        expectedCheckpoints = checkpoints,
+        expectedCheckpointVersion = checkpointVersion,
+        expectedLastCommitTimestamp = deltaVersions.max
+      )
+    }
+  }
+
+  test("getLogSegmentForVersion: corrupt _last_checkpoint with empty delta log") {
+    // listDeltaAndCheckpointFiles = Optional.empty()
+    val logSegmentOpt = new SnapshotManager().getLogSegmentForVersion(
+      logPath,
+      createMockTableClient(listFromFileList(Seq.empty)),
+      Optional.of(1),
+      Optional.empty()
+    )
+    assert(!logSegmentOpt.isPresent())
+  }
+}
+
+object SnapshotManagerSuite {
+
+  /**
+   * Create input function for createMockTableClient to implement listFrom from a list of
+   * file statuses.
+   * */
+  private def listFromFileList(files: Seq[FileStatus])(filePath: String): Seq[FileStatus] = {
+    files.filter(_.getPath.compareTo(filePath) >= 0).sortBy(_.getPath)
+  }
+
+  /**
+   * Create a mock {@link TableClient} to test log segment generation.
+   */
+  def createMockTableClient(listFromPath: String => Seq[FileStatus]): TableClient = {
+    new TableClient {
+      override def getExpressionHandler: ExpressionHandler = {
+        throw new UnsupportedOperationException("not supported for SnapshotManagerSuite tests")
+      }
+
+      override def getJsonHandler: JsonHandler = {
+        throw new UnsupportedOperationException("not supported for SnapshotManagerSuite tests")
+      }
+
+      override def getFileSystemClient: FileSystemClient =
+        createMockFileSystemClient(listFromPath)
+
+      override def getParquetHandler: ParquetHandler = {
+        throw new UnsupportedOperationException("not supported for SnapshotManagerSuite tests")
+      }
+    }
+  }
+
+  private def createMockFileSystemClient(
+    listFromPath: String => Seq[FileStatus]): FileSystemClient = {
+
+    new FileSystemClient {
+
+      override def listFrom(filePath: String): CloseableIterator[FileStatus] = {
+        val fileStatuses = listFromPath(filePath)
+
+        new CloseableIterator[FileStatus] {
```

Comment:
There is a utility method `toCloseableIterator`. May need to convert it to Java iterator.